<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1100</width>
    <height>750</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Openblocks Editor</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QMdiArea" name="mdiArea">
      <property name="viewMode">
       <enum>QMdiArea::ViewMode::TabbedView</enum>
      </property>
      <property name="documentMode">
       <bool>true</bool>
      </property>
      <property name="tabsClosable">
       <bool>true</bool>
      </property>
      <property name="tabsMovable">
       <bool>true</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1027</width>
     <height>30</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="actionSaveAs"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionCopy"/>
    <addaction name="actionCut"/>
    <addaction name="actionPaste"/>
    <addaction name="actionPasteInto"/>
    <addaction name="actionDelete"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QDockWidget" name="explorerWidget">
   <property name="windowTitle">
    <string>Explorer</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <widget class="ExplorerView" name="explorerView"/>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="propertiesWidget">
   <property name="windowTitle">
    <string>Properties</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <widget class="PropertiesView" name="propertiesView"/>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="outputWidget">
   <property name="windowTitle">
    <string>Output</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_3">
    <layout class="QVBoxLayout" name="verticalLayout_4">
     <item>
      <widget class="OutputTextView" name="outputTextView">
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QToolBar" name="fileTools">
   <property name="windowTitle">
    <string>File Tools</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionNew"/>
   <addaction name="actionOpen"/>
   <addaction name="actionSave"/>
   <addaction name="actionUndo"/>
   <addaction name="actionRedo"/>
  </widget>
  <widget class="QToolBar" name="transformTools">
   <property name="windowTitle">
    <string>Transform Tools</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionToolSelect"/>
   <addaction name="actionToolMove"/>
   <addaction name="actionToolScale"/>
   <addaction name="actionToolRotate"/>
   <addaction name="actionToggleSpace"/>
  </widget>
  <widget class="QToolBar" name="editTools">
   <property name="windowTitle">
    <string>Edit Tools</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionDelete"/>
   <addaction name="actionCopy"/>
   <addaction name="actionCut"/>
   <addaction name="actionPaste"/>
   <addaction name="actionPasteInto"/>
   <addaction name="actionGroupObjects"/>
   <addaction name="actionUngroupObjects"/>
  </widget>
  <widget class="QToolBar" name="snappingOptions">
   <property name="windowTitle">
    <string>Snapping Options</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionGridSnap1"/>
   <addaction name="actionGridSnap05"/>
   <addaction name="actionGridSnapOff"/>
  </widget>
  <widget class="QToolBar" name="simulationControls">
   <property name="windowTitle">
    <string>Simulation Controls</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionRunSimulation"/>
   <addaction name="actionPauseSimulation"/>
   <addaction name="actionStopSimulation"/>
  </widget>
  <widget class="QToolBar" name="surfaceTools">
   <property name="windowTitle">
    <string>Surface Tools</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionToolSmooth"/>
   <addaction name="actionToolGlue"/>
   <addaction name="actionToolWeld"/>
   <addaction name="actionToolStuds"/>
   <addaction name="actionToolInlets"/>
   <addaction name="actionToolUniversal"/>
   <addaction name="actionToolHinge"/>
   <addaction name="actionToolMotor"/>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>Sound Controls</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionToggleEditSounds"/>
  </widget>
  <widget class="QToolBar" name="commandBar">
   <property name="windowTitle">
    <string>Command Bar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>BottomToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <action name="actionAddPart">
   <property name="icon">
    <iconset>
     <normaloff>assets/icons/part.png</normaloff>assets/icons/part.png</iconset>
   </property>
   <property name="text">
    <string>Add Part</string>
   </property>
   <property name="toolTip">
    <string>Add a part to the workspace</string>
   </property>
  </action>
  <action name="actionNew">
   <property name="icon">
    <iconset theme="document-new"/>
   </property>
   <property name="text">
    <string>New</string>
   </property>
   <property name="toolTip">
    <string>New document</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="icon">
    <iconset theme="document-open"/>
   </property>
   <property name="text">
    <string>Open</string>
   </property>
   <property name="toolTip">
    <string>Open document</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset theme="document-save"/>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
   <property name="toolTip">
    <string>Save document</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionToolSelect">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="edit-select"/>
   </property>
   <property name="text">
    <string>Select Objects</string>
   </property>
   <property name="toolTip">
    <string>Select objects in the workspace</string>
   </property>
   <property name="shortcut">
    <string>1</string>
   </property>
  </action>
  <action name="actionToolMove">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="transform-move"/>
   </property>
   <property name="text">
    <string>Move Objects</string>
   </property>
   <property name="toolTip">
    <string>Move objects in the workspace</string>
   </property>
   <property name="shortcut">
    <string>2</string>
   </property>
  </action>
  <action name="actionToolScale">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="transform-scale"/>
   </property>
   <property name="text">
    <string>Scale Objects</string>
   </property>
   <property name="toolTip">
    <string>Scale objects in the workspace</string>
   </property>
   <property name="shortcut">
    <string>3</string>
   </property>
  </action>
  <action name="actionToolRotate">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="transform-rotate"/>
   </property>
   <property name="text">
    <string>Rotate Objects</string>
   </property>
   <property name="toolTip">
    <string>Rotate objects in the workspace</string>
   </property>
   <property name="shortcut">
    <string>4</string>
   </property>
  </action>
  <action name="actionToggleSimulation">
   <property name="icon">
    <iconset theme="media-playback-start"/>
   </property>
   <property name="text">
    <string>Start Simulation</string>
   </property>
   <property name="toolTip">
    <string>Start the simulation</string>
   </property>
   <property name="shortcut">
    <string>F5</string>
   </property>
  </action>
  <action name="actionGridSnap1">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="snap1"/>
   </property>
   <property name="text">
    <string>1-Stud Snapping</string>
   </property>
   <property name="toolTip">
    <string>Set grid snapping to 1 stud</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionGridSnap05">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="snap05"/>
   </property>
   <property name="text">
    <string>1/2-Stud Snapping</string>
   </property>
   <property name="toolTip">
    <string>Set grid snapping to 1/2 studs</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionGridSnapOff">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="snapoff"/>
   </property>
   <property name="text">
    <string>No Grid Snapping</string>
   </property>
   <property name="toolTip">
    <string>Turn grid snapping off</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionCopy">
   <property name="icon">
    <iconset theme="edit-copy"/>
   </property>
   <property name="text">
    <string>Copy</string>
   </property>
   <property name="toolTip">
    <string>Copy objects to clipboard</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionCut">
   <property name="icon">
    <iconset theme="edit-cut"/>
   </property>
   <property name="text">
    <string>Cut</string>
   </property>
   <property name="toolTip">
    <string>Cut objects into clipboard</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+X</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionPaste">
   <property name="icon">
    <iconset theme="edit-paste"/>
   </property>
   <property name="text">
    <string>Paste</string>
   </property>
   <property name="toolTip">
    <string>Paste objects from clipboard</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+V</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionPasteInto">
   <property name="icon">
    <iconset theme="edit-paste"/>
   </property>
   <property name="text">
    <string>Paste Into</string>
   </property>
   <property name="toolTip">
    <string>Paste objects from clipboard into selected object</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+V</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionDelete">
   <property name="icon">
    <iconset theme="edit-delete"/>
   </property>
   <property name="text">
    <string>Delete Object</string>
   </property>
   <property name="toolTip">
    <string>Delete selected objects</string>
   </property>
   <property name="shortcut">
    <string>Del</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionSaveModel">
   <property name="text">
    <string>Save Model to File</string>
   </property>
   <property name="toolTip">
    <string>Saves objects to file as XML model</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionInsertModel">
   <property name="text">
    <string>Insert Model from File</string>
   </property>
   <property name="toolTip">
    <string>Insert model from XML file</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionQuit">
   <property name="icon">
    <iconset theme="application-exit"/>
   </property>
   <property name="text">
    <string>Exit Openblocks Editor</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::QuitRole</enum>
   </property>
  </action>
  <action name="actionToggleSpace">
   <property name="icon">
    <iconset theme="space-local"/>
   </property>
   <property name="text">
    <string>Local</string>
   </property>
   <property name="toolTip">
    <string>Switch between local and world space transformations</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+L</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionSaveAs">
   <property name="icon">
    <iconset theme="document-save-as"/>
   </property>
   <property name="text">
    <string>Save As...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionToolStuds">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="surface-studs"/>
   </property>
   <property name="text">
    <string>Studs</string>
   </property>
   <property name="toolTip">
    <string>Studs</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionToolInlets">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="surface-inlets"/>
   </property>
   <property name="text">
    <string>Inlets</string>
   </property>
   <property name="toolTip">
    <string>Inlets</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionToolUniversal">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="surface-universal"/>
   </property>
   <property name="text">
    <string>Universal</string>
   </property>
   <property name="toolTip">
    <string>Universal</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionToolSmooth">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="surface-smooth"/>
   </property>
   <property name="text">
    <string>Smooth</string>
   </property>
   <property name="toolTip">
    <string>Smooth</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionToolWeld">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="surface-weld"/>
   </property>
   <property name="text">
    <string>Weld</string>
   </property>
   <property name="toolTip">
    <string>Weld</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionToolGlue">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="surface-glue"/>
   </property>
   <property name="text">
    <string>Glue</string>
   </property>
   <property name="toolTip">
    <string>Glue</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionToolHinge">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="surface-hinge"/>
   </property>
   <property name="text">
    <string>Hinge</string>
   </property>
   <property name="toolTip">
    <string>Hinge</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionToolMotor">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="surface-motor"/>
   </property>
   <property name="text">
    <string>Motor</string>
   </property>
   <property name="toolTip">
    <string>Motor</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionToggleEditSounds">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="audio-volume-high"/>
   </property>
   <property name="text">
    <string>Toggle Edit Sounds</string>
   </property>
   <property name="toolTip">
    <string>Toggles edit sound effects</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionRunSimulation">
   <property name="icon">
    <iconset theme="media-playback-start"/>
   </property>
   <property name="text">
    <string>Run</string>
   </property>
   <property name="toolTip">
    <string>Run simulation</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionPauseSimulation">
   <property name="icon">
    <iconset theme="media-playback-pause"/>
   </property>
   <property name="text">
    <string>Pause</string>
   </property>
   <property name="toolTip">
    <string>Pause simulation</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionStopSimulation">
   <property name="icon">
    <iconset theme="media-playback-stop"/>
   </property>
   <property name="text">
    <string>Stop</string>
   </property>
   <property name="toolTip">
    <string>Stop simulation</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionGroupObjects">
   <property name="icon">
    <iconset theme="object-group"/>
   </property>
   <property name="text">
    <string>Group Objects</string>
   </property>
   <property name="toolTip">
    <string>Group objects under a Model</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+G</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionUngroupObjects">
   <property name="icon">
    <iconset theme="object-ungroup"/>
   </property>
   <property name="text">
    <string>Ungroup Objects</string>
   </property>
   <property name="toolTip">
    <string>Ungroup objects inside selected Model</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+U</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About...</string>
   </property>
  </action>
  <action name="actionUndo">
   <property name="icon">
    <iconset theme="edit-undo"/>
   </property>
   <property name="text">
    <string>Undo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionRedo">
   <property name="icon">
    <iconset theme="edit-redo"/>
   </property>
   <property name="text">
    <string>Redo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Y</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ExplorerView</class>
   <extends>QTreeView</extends>
   <header>panes/explorerview.h</header>
  </customwidget>
  <customwidget>
   <class>PropertiesView</class>
   <extends>QTreeView</extends>
   <header>panes/propertiesview.h</header>
  </customwidget>
  <customwidget>
   <class>OutputTextView</class>
   <extends>QTextEdit</extends>
   <header>panes/outputtextview.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
